### PR TITLE
Test cases for Whois API

### PR DESCRIPTION
**Pull request by sinchanagoudar**
Written the test cases for WhoIs API , It includes home, pricing, apiinfo and documentation

### CODE REVIEW

File: cypress/e2e/WhoIsApi/apiinfo.cy.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { WHOISAPI_URL } from '../../resources/data';
+
+describe('Who Is API Info page', () => {
+    before(() => {
+      cy.visit(WHOISAPI_URL);
+    });
+
```
Comment (AngadCel) [inline cypress/e2e/WhoIsApi/apiinfo.cy.js:7]: remove the extra space

File: cypress/e2e/WhoIsApi/pricing.cy.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    it('7. Test to check price of "Starter plan" is present', () => {
+        cy.componentVisiblityCheck(':nth-child(2) > .card > .card-header > .mb-3 > .plan-price', '$4.99');
+    });
+
+    it('8. Test to check features of "Starter plan" is present', () => {
+        cy.componentVisiblityCheck('#pricing > .row > :nth-child(2) > .card > .card-body > :nth-child(1) > .media-body', '150,000\nRequests / Monthly');
+        cy.componentVisiblityCheck(':nth-child(2) > .card > .card-body > :nth-child(2) > .media-body', 'Standard Support');
+    });
+
```
Comment (AngadCel) [inline cypress/e2e/WhoIsApi/pricing.cy.js:42]: remove extra space

File: cypress/e2e/WhoIsApi/home.cy.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    });
+
+    it('10. Test to check "Related Products" container is present', () => {
+        cy.componentVisiblityCheck('.h1', 'Related Products');
+    });
+
+    it('11. Test to check footer is present', () => {
+        cy.componentVisiblityCheck('footer.bg-primary');
+    });
+
```
Comment (AngadCel) [inline cypress/e2e/WhoIsApi/home.cy.js:53]: image
missing test case for check the click event on "check the documentation"

File: cypress/e2e/WhoIsApi/apiinfo.cy.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    it('9. Test to check "How it works" subtitle is present', () => {
+        cy.checkTextVisibility('How it works');
+    });
+
+    it('10. Test to check "Related Products" subtitle is present', () => {
+        cy.checkTextVisibility('Related Products');
+    });
+
```
Comment (AngadCel) [inline cypress/e2e/WhoIsApi/apiinfo.cy.js]: test case missing for check visibility for "Which TLDs are supported?"

Comment (AngadCel) [inline cypress/e2e/WhoIsApi/apiinfo.cy.js]: image
click event is missing

File: cypress/e2e/WhoIsApi/documentation.cy.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    });
+
+    it('12. Test to check "Error Codes" hyperlink is present', () => {
+        cy.componentVisiblityCheck('.sticky-top > ul > :nth-child(4) > a', 'Error Codes');
+    });
+  
+    it('13. Test to click on "Error Codes" hyperlink and check "Error Codes" title is present', () => {
+        cy.get('.sticky-top > ul > :nth-child(4) > a').click();
+    });
+
```
Comment (AngadCel) [inline cypress/e2e/WhoIsApi/documentation.cy.js:95]: image
click event missed

Comment (AngadCel) [inline cypress/e2e/WhoIsApi/documentation.cy.js:95]: image

click event missed

Comment (AngadCel) [inline cypress/e2e/WhoIsApi/documentation.cy.js:64]: image
missing test case

File: cypress/e2e/WhoIsApi/pricing.cy.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    it('13. Test to check price of "Custom plan" is present', () => {
+        cy.componentVisiblityCheck(':nth-child(4) > .card > .card-header > .mb-3 > .font-size-4', 'Volume');
+    });
+  
+    it('14. Test to check features of "Custom plan" is present', () => {
+        cy.componentVisiblityCheck('#pricing > .row > :nth-child(4) > .card > .card-body > .media > .media-body', 'Any requests volume you need');
+    });
+
```
Comment (AngadCel) [inline cypress/e2e/WhoIsApi/pricing.cy.js:67]: image
missing test case for click on contact us

File: cypress/e2e/WhoIsApi/apiinfo.cy.js (hunk 1 above)
Comment (sinchanagoudar) [reply cypress/e2e/WhoIsApi/apiinfo.cy.js:7]: done

File: cypress/e2e/WhoIsApi/apiinfo.cy.js (hunk 4 above)
Comment (sinchanagoudar) [reply cypress/e2e/WhoIsApi/apiinfo.cy.js]: Added the test case

Comment (sinchanagoudar) [reply cypress/e2e/WhoIsApi/apiinfo.cy.js]: Added

File: cypress/e2e/WhoIsApi/home.cy.js (hunk 3 above)
Comment (sinchanagoudar) [reply cypress/e2e/WhoIsApi/home.cy.js:53]: After clicking on the link the route is not getting change , not valid test case

File: cypress/e2e/WhoIsApi/pricing.cy.js (hunk 2 above)
Comment (sinchanagoudar) [reply cypress/e2e/WhoIsApi/pricing.cy.js:42]: Done

File: cypress/e2e/WhoIsApi/pricing.cy.js (hunk 6 above)
Comment (sinchanagoudar) [reply cypress/e2e/WhoIsApi/pricing.cy.js:67]: done

File: cypress/e2e/WhoIsApi/documentation.cy.js (hunk 5 above)
Comment (sinchanagoudar) [reply cypress/e2e/WhoIsApi/documentation.cy.js:95]: done

File: cypress/e2e/WhoIsApi/apiinfo.cy.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        cy.get('[href="/account"]').click();
+        cy.navigateUrlwithCookies('https://apilayer.com/?signin');
+        cy.navigateUrlwithCookies(WHOISAPI_BASE_URL);
+        cy.get('#details-tab > .d-md-flex > span').click();
+
+    });
+
+    it('14. Test to click on "discover potential domain names" hyperlink', () => {
+        cy.get('[href="/account"]').click();
+        cy.navigateUrlwithCookies('https://apilayer.com/marketplace/domain_discovery-api');
```
Comment (NaveenK15) [inline cypress/e2e/WhoIsApi/apiinfo.cy.js]: base URL

File: cypress/e2e/WhoIsApi/documentation.cy.js [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+    it('11. Test to click on "Endpoints" hyperlink and check "Endpoints" title is present', () => {
+        cy.get('.sticky-top > ul > :nth-child(2) > a').click();
+        cy.componentVisiblityCheck(':nth-child(9) > a', 'Endpoints');
+    });
+
+    it('12. Test to check the "Subscribe" button is present', () => {
+        cy.componentVisiblityCheck('#subscribeButton');
+    });
+
+    it('13. Test to check by clicing the "Subscribe" button', () => {
```
Comment (NaveenK15) [inline cypress/e2e/WhoIsApi/documentation.cy.js]: clicking

Comment (sinchanagoudar) [reply cypress/e2e/WhoIsApi/documentation.cy.js]: Done

**Pull request merged**